src/dotnet-install.sh

```diff
@@ -921,10 +921,16 @@ get_http_header_wget() {
     local remote_path="$1"
     local disable_feed_credential="$2"
     local wget_options="-q -S --spider --tries 5 "
-    # Store options that aren't supported on all wget implementations separately.
-    local wget_options_extra="--waitretry 2 --connect-timeout 15 "
+
+    local wget_options_extra=''
     local wget_result=''
 
+    # Test for options that aren't supported on all wget implementations.
+    wget -h 2>&1 | grep "waitretry" >/dev/null && wget -h 2>&1 | grep "connect-timeout" >/dev/null
+    if [ $? = 0 ]; then
```

Comment:
We have recently fixed an issue with checking the results with `$?` here https://github.com/dotnet/install-scripts/pull/260.

Can we instead do something like:
```
if ! wget -h 2>&1 | grep .... ; then
```
instead. This will likely fail with `sudo` otherwise.

Reply:
@danmoseley,
I will take it from here.

Thank you for fixing the main issue.

Reply:
@bekir-ozturk ,

the fix is ready, I have tested it.
Please review